### DISCUSSION TITLE
Feature: allow to modify routes at runtime

### DISCUSSION THREAD

**Discussion by egoist** [remix-run/remix#9266] [Proposals]
I need this primarily for multi-tenancy support in my app, i.e. allow my users to use a subdomain of my website or bring their own custom domain

Normally this can be implemented by mapping different hostname to different routes, but this is not possible since Remix routes are generated at build time.

My proposal is to introduce a new hook in `entry.server.ts` to allow modifying the routes at runtime:

```ts
export const handleRoutes = (request: Request, routes: RouteManifest<EntryRoute>) => {
  return getNewRoutesSomehow(request.headers.host, routes)
}
```

This feature would also fix https://github.com/remix-run/remix/discussions/2886

**Comment by brophdawg11** [comment]
This should be handled by https://github.com/remix-run/react-router/discussions/11113
